src/_P145_MQxxx.ino

```diff
@@ -0,0 +1,282 @@
+      Device[deviceCount].Ports = 0;
+      Device[deviceCount].PullUpOption = false;
+      Device[deviceCount].InverseLogicOption = false;
+      Device[deviceCount].FormulaOption = true;
+      Device[deviceCount].ValueCount = 1;
+      Device[deviceCount].SendDataOption = true;
+      Device[deviceCount].TimerOption = true;
+      Device[deviceCount].GlobalSyncOption = true;
+      break;
+    }
+    
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_145);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_145));
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      // FormSelector with all predefined "Sensor - Gas" options
+      String options[P145_MAXTYPES] = {};
+      int optionValues[P145_MAXTYPES] = {};
+      int x = P145_data_struct::getNbrOfTypes();
+      if (x > P145_MAXTYPES) 
+      {
+        x = P145_MAXTYPES;
+      }
+      for (int i=0; i<x; i++)
+      {
+        options[i] = (String)P145_data_struct::getTypeName(i) + (String)(F(" - ")) + (String)P145_data_struct::getGasName(i);
+        optionValues[i] = i; 
+      }
+      addFormSelector(F("Sensor type"), F("plugin_145_sensor"), x, options, optionValues, PCONFIG_SENSORT);
+
+# ifdef ESP32
+      // Analog input selection
+      addRowLabel(F("Analog Pin"));
+      addADC_PinSelect(AdcPinSelectPurpose::ADC_Touch_HallEffect, F("taskdevicepin1"), CONFIG_PIN1);
+# endif // ifdef ESP32
+
+      addFormTextBox(F("Load Resistance"), F("plugin_145_RLOAD"), String(PCONFIG_FLOAT_RLOAD), 33);
+      addUnit(F("Ohm"));
+      addFormTextBox(F("R Zero"), F("plugin_145_RZERO"), String(PCONFIG_FLOAT_RZERO), 33);
+      addUnit(F("Ohm"));
+      addFormTextBox(F("Reference Level"), F("plugin_145_REFLEVEL"), String(PCONFIG_FLOAT_REF), 33);
+      addUnit(F("ppm"));
+      P145_data_struct *P145_data = static_cast<P145_data_struct *>(getPluginTaskData(event->TaskIndex));
+      if (P145_data != nullptr)
+      {
+        float calVal = P145_data->getCalibrationValue();
+        if (calVal > 0.0) 
+        {
+          addFormNote((String)F("Current measurement suggests Rzero= ") + (String)calVal);
+        }
+      }
+
+      addFormSeparator(2);
+      // Auto calibration and Temp/hum compensation flags are bitfields in PCONFIG_FLAGS
+      bool compensate = PCONFIG_FLAGS & 0x0001;
+      bool calibrate = (PCONFIG_FLAGS >> 1) & 0x0001;
+      addFormCheckBox(F("Enable automatic calibration"), F("plugin_145_enable_calibrarion"), calibrate);
+      addFormCheckBox(F("Enable temp/humid compensation"), F("plugin_145_enable_compensation"), compensate);
+      addFormNote(F("If this is enabled, the Temperature and Humidity values below need to be configured."));
+      // temperature
+      addHtml(F("<TR><TD>Temperature:<TD>"));
+      addTaskSelect(F("plugin_145_temperature_task"), PCONFIG_TEMP_TASK);
```

Comment:
This `addHtml(...);` can be replaced by `addRowLabel(F("Temperature"));`, thus using existing functions and shorter strings.

and the variable names like `"plugin_145_temperature_task"` can do without the `plugin_145_` prefix (there will be only 1 page on screen at any moment), and preferably all lower-case (above), also saving a few bytes in .bin size (have started a PR cleaning/compacting all those unneeded prefixes and other not really needed long strings for .bin size reason...)